test/setup/deployers/operator-olm.ts

```diff
@@ -33,14 +35,37 @@ async function waitToDeployKubernetesOperator(namespace: string): Promise<void>
 async function deployKubernetesMonitor(
   _imageOptions: IImageOptions,
 ): Promise<void> {
-    await kubectl.applyK8sYaml('./test/fixtures/operator/operator-source-k8s.yaml');
-    await kubectl.applyK8sYaml('./test/fixtures/operator/installation-k8s.yaml');
-
-    // Await for the Operator to become available, only then
-    // the Operator can start processing the custom resource.
-    await kubectl.waitForDeployment('snyk-operator', 'marketplace');
-    await kubectl.waitForCRD('snykmonitors.charts.helm.k8s.io');
-    await kubectl.applyK8sYaml('./test/fixtures/operator/custom-resource-k8s.yaml');
-    await waitToDeployKubernetesOperator('marketplace');
-    await kubectl.waitForDeployment('snyk-operator', 'marketplace');
+  const overriddenOperatorSource = 'snyk-monitor-catalog-source.yaml';
+  createTestOperatorSource(overriddenOperatorSource);
+  await kubectl.applyK8sYaml(overriddenOperatorSource);
+  await kubectl.applyK8sYaml('./test/fixtures/operator/installation-k8s.yaml');
+
+  // Await for the Operator to become available, only then
+  // the Operator can start processing the custom resource.
+  await kubectl.waitForDeployment('snyk-operator', 'marketplace');
+  await kubectl.waitForCRD('snykmonitors.charts.helm.k8s.io');
+  await kubectl.applyK8sYaml('./test/fixtures/operator/custom-resource-k8s.yaml');
+  await waitToDeployKubernetesOperator('marketplace');
+  await kubectl.waitForDeployment('snyk-operator', 'marketplace');
+}
+
+function createTestOperatorSource(newYamlPath: string): void {
+  console.log('Creating YAML CatalogSource...');
+  const operatorVersion =
+    process.env.OPERATOR_VERSION ?? readFileSync('./.operator_version', 'utf8');
```

Comment:
is`readFileSync('./.operator_version', 'utf8')` for running the tests locally?

Reply:
Yesss, since we have a crazy mix of shell, python, npm scripts to automate all of it to a single line 😵